pkg/apis/build/v1beta1/buildrun_types.go

```diff
@@ -0,0 +1,394 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// BuildRunDomain is the domain used for all labels and annotations for this resource
+	BuildRunDomain = "buildrun.shipwright.io"
+
+	// LabelBuildRun is a label key for BuildRuns to define the name of the BuildRun
+	LabelBuildRun = BuildRunDomain + "/name"
+
+	// LabelBuildRunGeneration is a label key for BuildRuns to define the generation
+	LabelBuildRunGeneration = BuildRunDomain + "/generation"
+)
+
+type ReferencedBuild struct {
+	// Build refers to an embedded build specification
+	//
+	// +optional
+	Build *BuildSpec `json:"spec,omitempty"`
+
+	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
```

Comment:
Standalone BuildRun's don't have a build name.

```suggestion
	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
	//
	// +optional
	Name *string `json:"name,omitempty"`
```